### PR TITLE
PLANNER-1380: Add Nurse Rostering benchmark

### PR DESCRIPTION
**Pull request by Fonifan**
Depends on merging of [this PR](https://github.com/kiegroup/kie-benchmarks/pull/38).

### CODE REVIEW

File: optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.optaplanner.core.config.heuristic.selector.move.composite.UnionMoveSelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.move.factory.MoveListFactoryConfig;
+import org.optaplanner.core.config.localsearch.LocalSearchPhaseConfig;
+import org.optaplanner.core.config.localsearch.decider.acceptor.AcceptorConfig;
+import org.optaplanner.core.config.localsearch.decider.forager.LocalSearchForagerConfig;
+import org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig;
+import org.optaplanner.core.config.solver.SolverConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+
+public class NurseRosteringBenchmark extends AbstractPlannerBenchmark<NurseRoster> {
```
[the remark this answers is not in the document]
Comment (Fonifan) [reply optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java]: After several experiments, I found out that 25 is optimal count of warmup iterations.

[the remark this answers is not in the document]
Comment (rsynek) [reply optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java]: Could you please explain how you did the experiments?

[the remark this answers is not in the document]
Comment (Fonifan) [reply optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java]: I ran benchmarks with following parameters : 10 measurement iterations, 5 forks and 10,25,35 warmup iterations. Following three comments will be results(every combination was run twice)

[the remark this answers is not in the document]
Comment (Fonifan) [reply optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java]: Parameters | Values | Error | Error %
-- | -- | -- | --
Forks | 5 |   |  
Measurement | 10 |   |  
Warmup | 10 |   |  
  |   |   |  
  |   |   |  
SPRINT | 2198.640 | 408.670 | 18,59 %
MEDIUM | 546.292 | 75.595 | 13,84 %
LONG | 867.603 | 158.568 | 18,28 %
  |   |   |  
SPRINT | 1604.298 | 138.054 | 8,61 %
MEDIUM | 503.502 | 78.482 | 15,59 %
LONG | 576.386 | 66.757 | 11,58 %

[the remark this answers is not in the document]
Comment (Fonifan) [reply optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java]: Parameters | Values | Error | Error %
-- | -- | -- | --
Forks | 5 |   |  
Measurement | 10 |   |  
Warmup | 25 |   |  
  |   |   |  
  |   |   |  
SPRINT | 1714.958 | 154.962 | 9,04 %
MEDIUM | 434.848 | 29.562 | 6,80 %
LONG | 501.709 | 23.519 | 4,69 %
  |   |   |  
SPRINT | 1454.798 | 72.237 | 4,97 %
MEDIUM | 425.554 | 30.280 | 7,12 %
LONG | 551.684 | 38.503 | 6,98 %

[the remark this answers is not in the document]
Comment (Fonifan) [reply optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java]: Parameters | Values | Error | Error %
-- | -- | -- | --
Forks | 5 |   |  
Measurement | 10 |   |  
Warmup | 35 |   |  
  |   |   |  
  |   |   |  
SPRINT | 1541.030 | 109.820 | 7,13 %
MEDIUM | 445.811 | 43.132 | 9,67 %
LONG | 532.507 | 36.448 | 6,84 %
  |   |   |  
SPRINT | 1521.963 | 68.281 | 4,49 %
MEDIUM | 435.468 | 20.343 | 4,67 %
LONG | 513.731 | 38.492 | 7,49 %

[the remark this answers is not in the document]
Comment (rsynek) [reply optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/realworld/nurserostering/NurseRosteringBenchmark.java]: Looks good, let's repeat this experiment in a stable environment and find the sweet spot.

**Pull request merged**